lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,238 @@
+    var clipDetails = await getCurrentPodcastClipDetails();
+    int currentClipDuration = clipDetails.clipDuration;
+
+//If  duration is less than 10 seconds return
+    if (_maxClipDuration - currentClipDuration <= 0) {
+      return;
+    }
+    if ((clipDetails.endTimeStamp + Duration(seconds: _initialSeconds)) >
+        clipDetails.episodeLength) {
+      return;
+    }
+    int incrementedClipDuration = 0;
+    Duration incrementedClipEndTime = Duration(seconds: 0);
+    if (currentClipDuration % 10 != 0) {
+      incrementedClipDuration =
+          currentClipDuration - currentClipDuration % 10 + _initialSeconds;
+      incrementedClipEndTime = Duration(
+          seconds:
+              clipDetails.startTimeStamp.inSeconds + incrementedClipDuration);
+    } else {
+      incrementedClipDuration = currentClipDuration + _initialSeconds;
+      incrementedClipEndTime = Duration(
+          seconds: clipDetails.endTimeStamp.inSeconds + _initialSeconds);
+    }
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: incrementedClipDuration,
+        endTimeStamp: incrementedClipEndTime));
+  }
+
+  decrementCount() async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+    int currentCLipDuration = clipDetails.clipDuration;
+
+    //return if duration needs to be greator than 10 seconds
+    if (currentCLipDuration <= _initialSeconds) {
+      return;
+    }
+    int decrementedClipDuration = 0;
+    Duration decrementedClipEndTime = Duration(seconds: 0);
+
+    if (currentCLipDuration % 10 != 0) {
+      decrementedClipDuration = currentCLipDuration - currentCLipDuration % 10;
+      decrementedClipEndTime = Duration(
+          seconds:
+              clipDetails.endTimeStamp.inSeconds - currentCLipDuration % 10);
+    } else {
+      decrementedClipDuration = currentCLipDuration - _initialSeconds;
+      decrementedClipEndTime =
+          clipDetails.endTimeStamp - Duration(seconds: _initialSeconds);
+    }
+
+    _clipDetailsBehaviourSubject.add(clipDetails.copy(
+        clipDuration: decrementedClipDuration,
+        endTimeStamp: decrementedClipEndTime));
+  }
+
+  Future<String> _getAudioClipPath(
+      {PodcastClipDetailsModel clipDetails}) async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+
+    String episodeUrl = clipDetails.episodeDetails.contentUrl;
+    String audioClipPath = await initClipDirectory(isVideoClip: false);
+    audioClipPath = audioClipPath + '/clip.mp3';
+
+    //Getting the raw audio clip
+    String clippedAudioCommand =
+        '-i "$episodeUrl" -ss ${clipDetails.startTimeStamp.inSeconds} -to ${clipDetails.endTimeStamp.inSeconds}  -acodec copy $audioClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedAudioCommand);
+
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return audioClipPath;
+    } else {
+      return null;
```

Comment:
why not just throw error? returning null will swallow the real error.

Reply:
Added throw.